lib/model.js

```diff
@@ -2215,22 +2215,25 @@ class Model {
    * @param  {Boolean}      [options.validate=true] Run validations before the row is inserted
    * @param  {Array}        [options.fields=Object.keys(this.attributes)] The fields to insert / update. Defaults to all changed fields
    * @param  {Boolean}      [options.hooks=true]  Run before / after upsert hooks?
+   * @param  {Boolean}      [options.returning=false] Append RETURNING * to get back auto generated values (Postgres only)
    * @param  {Transaction}  [options.transaction] Transaction to run query under
    * @param  {Function}     [options.logging=false] A function that gets executed while running the query to log the sql.
    * @param  {Boolean}      [options.benchmark=false] Pass query execution time in milliseconds as second argument to logging function (options.logging).
    * @param  {String}       [options.searchPath=DEFAULT] An optional parameter to specify the schema search_path (Postgres only)
    *
-   * @return {Promise<created>} Returns a boolean indicating whether the row was created or updated.
+   * @return {Promise<created>} Returns a boolean indicating whether the row was created or updated. For Postgres with (options.returning=true), it returns record and created boolean with signature `<Model, created>`.
```

Comment:
In v5, we can set this `Promise<record, created>` by setting `returning=true` by default. One caveat is `null` is now first argument for other dialects. May be should set this as ``Promise<created, record>``, but that doesn't match with other method's API. Any ideas?

Reply:
I have added support for MSSQL as well, so now may be current order make sense.